.gitlab-ci.yml

```diff
@@ -0,0 +1,36 @@
+include:
+  - project: 'analytics/artificial-intelligence/ai-platform/aip-infrastructure/ci-templates/ci-cd-template'
+    ref: &include_ref 'alexla/AIP-4600'  # TODO AIP-4600 Replace with v2
+    file: '/blocks/python.yml'
+
+stages:
+  - build
+
+build:publish:
+  extends: .aip_python_debian_image
+  stage: build
+  variables:
+    PY_LIBRARY_NAME: 'zillow-kfp'
+    KFP_VERSION_PATH: "sdk/python/kfp/__init__.py"
+  script:
+  - KFP_VERSION=$(cat $KFP_VERSION_PATH | sed -nr "s/^__version__ = ['\"]([^'\"]*)['\"]$/\1/p")
+  - |
+    if [ "${CI_COMMIT_BRANCH}" = "${CI_DEFAULT_BRANCH}" ]; then
+      PY_LIBRARY_VERSION="${KFP_VERSION}.${CI_PIPELINE_IID}"
+    else
+      PY_LIBRARY_VERSION="0.0.${CI_PIPELINE_IID}+${CI_COMMIT_SHORT_SHA}"
+    fi
+  # Now write back the zillow-kfp version back to the original location we found the original.
+  - sed -i "s/\(__version__ = ['\"]\)[^'\"]*\(['\"]\)/\1${PY_LIBRARY_VERSION}\2/" $KFP_VERSION_PATH
```

Comment:
I noticed we're duplicating this regex expression with the Metaflow publishing PR. Would it be worth storing in the `ci-cd templates` and giving it an easy to read name (e.g. `substitute_version_path_with_library_version`)?

Reply:
Yes and no. Whereas `kfp` exposes this as `__version__` in the `kfp/__init__.py`, `metaflow` exposes this as `version` in `setup.py` so I figured it was just different enough to not bother.

I think we likely need to spend some time making a paved path CICD template for libraries (trust me I wanted to do this in this round of PRs) but I decided given the scope of the current changes to not expand things to just move on with the `v3.9` upgrade 😅  Definitely something I'd like to investigate in the near future.